Eat_and_Greet/src/screens/PostEvent/index.tsx

```diff
@@ -1,18 +1,72 @@
-import { Text, View } from "react-native";
+import React, { useState } from "react";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { Text, View, Image, Alert } from "react-native";
 import { StatusBar } from "expo-status-bar";
-
-import { Toolbar } from "../../comps/Toolbar/toolbar";
+import { db } from "../../firebase/config";
+import { collection, addDoc, getDocs, query, where } from "firebase/firestore";
 
 import styles from "../../style";
-import React from "react";
-import RoundedButton from "../../comps/RoundedButton/RoundedButton";
-import { Image } from "react-native";
-import { SafeAreaView } from "react-native-safe-area-context";
-
 import postStyle from "./index.styles";
+
+import { Toolbar } from "../../comps/Toolbar/toolbar";
+import RoundedButton from "../../comps/RoundedButton/RoundedButton";
 import TextBox from "../../comps/Textbox/textbox";
 
 export default function PostEvent() {
+  const [Title, setTitle] = useState("");
+  const [Category, setCategory] = useState("");
+  const [EventDate, setEventDate] = useState("");
+  const [StartTime, setStartTime] = useState("");
+  const [EndTime, setEndTime] = useState("");
+
+  const addPostToDB = async () => {
+    if (Title.length > 0) {
```

Comment:
Is there a way to do this without 5 ifs,  trys excepts, etc? This seems overly complex and seems to be done for error logging. Could you have it confirming nothing in a list has length < 0 and error log according to what does? Then at the end you could say "Missing ___, ___, ___" instead of being dependent on the previous succeeding. IE if there isnt a start time or end time, it should say both are missing.